### PR TITLE
Add https support to ansible: role, plugin and module

### PR DESCRIPTION
**Pull request by nessiLahav**
#### What does this PR do?
* Add HTTPS support to ansible: role, plugin and module
* The Summon.sh still not working with HTTPS I waiting for @doodlesbykumbi  to fix the api-go to support custom certificate
#### Where should the reviewer start?
* Run test.sh on your environment and check it is finish without errors 
#### How should this be manually tested?
* Add bash command to test.sh and connect inside the containers if you need

#### What are the relevant tickets?
Ansible board: https://waffle.io/cyberark/ansible-role-conjur
#8 HTTPS connection with Conjur server

#### Reviewers
@jvanderhoof 
@dustinmm80 
@jeniaSakirko 
@orenbm 
#### Questions:
- Can we make a blog post, a video, an animated GIF out of this?
- Is this explained in a training tutorial?
- Does the knowledge base need an update?
- Is there a demo?

### CODE REVIEW

File: lookup_plugins/retrieve_conjur_variable.py
Comment (orenbm) [inline lookup_plugins/retrieve_conjur_variable.py]: I don't think this comment should be in the actual code as this code is open source and this is the actual product the customer will get. The proxy server is just for our testing.

File: lookup_plugins/retrieve_conjur_variable.py
Comment (orenbm) [inline lookup_plugins/retrieve_conjur_variable.py]: Also here, I think (correct me if I'm wrong) that the hostname verification dosn't work in our environment but in an actual environmnet of the customer it should work, right? Isn't this sort of a code smell?

File: tests/Dockerfile.nginx
Comment (orenbm) [inline tests/Dockerfile.nginx]: I believe `Maintainer` is deprecated and that `LABEL` replaced it. I didn't read a lot about it, maybe you'll find something here:
https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#label

File: tests/docker-compose.yml
Comment (orenbm) [inline tests/docker-compose.yml]: No newline :)

**Review by orenbm — CHANGES_REQUESTED** [review]

File: tests/Dockerfile.nginx
Comment (orenbm) [inline tests/Dockerfile.nginx]: Can you add `rm -rf /var/lib/apt/lists/*` at the end of this command? It makes the cached image smaller and more light-weight.

**Review by orenbm — CHANGES_REQUESTED** [review]

File: tests/Dockerfile.nginx
Comment (jvanderhoof) [reply tests/Dockerfile.nginx]: Nice spot @orenbm!  I'll update it in the AnsibleFest demo as well. I never liked the syntax for company projects.

File: library/retrieve_conjur_variable.py
Comment (jvanderhoof) [inline library/retrieve_conjur_variable.py:5]: I believe we should support both HTTP & HTTPS connections.  As a developer working locally, running HTTPS can be very challenging to setup. We can use the Role to configure HTTPS connections with the `conjur_url`, `ssl_certificate`, and `validate_certs` arguments.  If `conjur_url` starts with `https`, we'll need to use `HTTPSConnection`.  If `validate_certs` if false, we can use insecure ssl connections.  The `ssl_certificate` attribute takes an SSL cert to use for HTTPS connections.

What do you think about expanding the `conjur.conf.j2 template to include these additional connection configuration scenarios?  It would allow the plugin and module to make use of the connection setup laid down by the role.

File: lookup_plugins/retrieve_conjur_variable.py
Comment (jvanderhoof) [reply lookup_plugins/retrieve_conjur_variable.py]: I agree. Using a proxy should be an implementation detail left to the end user.  The role should setup the connect as desired by the user, but not care about what that setup looks like.

File: lookup_plugins/retrieve_conjur_variable.py
Comment (jvanderhoof) [reply lookup_plugins/retrieve_conjur_variable.py]: If we allow four forms of connection:
1. HTTP 
2. HTTPS
3. HTTPS with provided CA cert
4. HTTPS without cert validation

this shouldn't be an issue.

Comment (nessiLahav) [reply lookup_plugins/retrieve_conjur_variable.py]: Oren: Fixed by certificate v3 with alt name = localhost, I will delete the workaround from the plugin

Jason: 1,2,4 wil be fix in another branch I am waiting for the previous PRs to be merge
Jason: 3 decided with Inbal to work on it in low priority

File: library/retrieve_conjur_variable.py
Comment (nessiLahav) [reply library/retrieve_conjur_variable.py:5]: Yeah, I already implemented the same logic you wrote :) 
 I am waiting for the previous PRs to be merge to avoid conflicts, when it we be bone I will create the PR with HTTPS & HTTP support 

For meanwhile you can take a look on the branch : Fix_ssl_issue_32

**Review by orenbm — APPROVED** [review]

**Pull request merged**
